plugin-admin/generate_plugins.js

```diff
@@ -0,0 +1,74 @@
+// Generate plugin data file from npm node_modules and previous manually created plugin data file
+const fs = require('fs')
+const jsonic = require('jsonic')
+
+let pluginData = {}
+
+let pkgjson = require('./package.json')
+const pkgmap = Object.keys(pkgjson.dependencies).reduce(
+  (pkgmap, dep) => (
+    (pkgmap[dep] = require('./node_modules/' + dep + '/package.json')), pkgmap
+  ),
+  {}
+)
+
+Object.values(pkgmap).forEach((pkg) => {
+  // npm fields: name, title, org_repo, desc
+  let title_match = pkg.name.match(/^@?seneca[-|/]([a-z|-]+)/)
+  let title = title_match ? title_match[1] : pkg.name
+  let org_repo = pkg.name
+  if (null != pkg.repository) {
+    let orgrepo_match = pkg.repository.url.match(
+      /[a-z]+:\/\/github.com\/([a-z]+\/[a-z|-]+).git/
+    )
+    org_repo = orgrepo_match ? orgrepo_match[1] : pkg.name
+  }
+  pluginData[pkg.name] = {
+    title: title,
+    org_repo: org_repo,
+    desc: pkg.description,
+  }
+})
+
+// ejs fields: badges, badges urls
+let ejs = fs
+  .readFileSync('../src/pages/plugins-2023/plugins_def.ejs')
+  .toString()
+ejsStr = ejs.slice(16, -2)
+let eplugins = jsonic(ejsStr)
+pluginGroups = Object.keys(eplugins)
+pluginGroups.forEach((groupName) => {
+  let group = eplugins[groupName]
+  pluginNames = Object.keys(group)
+  pluginData[groupName] = { XdescX: '' }
+  pluginNames.forEach((plugin) => {
+    if ('XdescX' == plugin) {
```

Comment:
A variable I created to add a description for each family, something that does not exist anywhere else (github, npm etc). I named it XdescX so that it wouldn't accidentally match something else - it's not a fixed name

Reply:
but a plugin family is not itself a plugin, so why would this be needed?
(in general, avoid mixing data types)

Reply:
Refactor on the way 👍